Hunting Queries/DnsEvents/DNS_HighNXDomainCount.txt

```diff
@@ -1,24 +1,82 @@
 // Name: DNS high NXDomain count
 // Id:543e1ec6-ee5e-4368-aaa6-405f0551ba5c
-// 
-// Description: clients with a high NXDomain count could be indicative of a DGA (cycling through possible C2 domains
+// Author: Vani Asawa
+//
+// Description: Clients with a high NXDomain count could be indicative of a DGA (cycling through possible C2 domains
 // where most C2s are not live). 
 //
 // DataSource: #DnsEvents
 //
 // Tactics: #C2, #Exfiltration
 //
 // NXDomain events, with some likely uninteresting records filtered out
-let nxDomainDnsEvents = DnsEvents 
+//
+let nxDomainDnsEvents = Dns
 | where ResultCode == "3" 
 | where QueryType in ("A", "AAAA")
 | where ClientIP != "127.0.0.1"
 | where Name !contains "/"
-| where Name contains ".";
-// Query for ClientIPs with high NXDomain counts
+| where Name contains "."
+| extend mytld = tostring(split(Name, '.')[-1])
+| extend truncatedDomain = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+| where mytld !in ("arris","ati","virtusa","unknowndomain","onion","corp","domain","local","localdomain","host","home","gateway","lan","services","hub","domain.name","WirelessAP","Digicom-ADSL","OpenDNS","dlinkrouter","Dlink","ASUS","device","router","Belkin","DHCP","Cisco");
```

Comment:
let's filter (cheap operation) before the more expensive strcat/split operations on the previous line